src/components/pdfPage/PdfPage.tsx

```diff
@@ -127,7 +127,11 @@ const PdfPage: FC<IPdfPageProps> = (props) => {
     setPageScale(scale);
   }, [pageViewport, scale, props.width, props.height]);
   const isBboxSelected = (bbox: IBbox) => props.activeBboxIndex === bbox.index;
-  const isRelated = (bbox: IBbox) => props.groupId ? bbox.groupId === props.groupId && !isBboxSelected(bbox) : false;
+  const isRelated = (bbox: IBbox) => {
+    const [, activeTest, activeId] = props?.groupId?.split('-') || [];
+    const [, bboxTest, bboxId] = bbox?.groupId?.split('-') || [];
+    return props.groupId ? (activeTest !== bboxTest && activeId === bboxId || bbox.groupId === props.groupId) && !isBboxSelected(bbox) : false;
```

Comment:
Can't we just check the `activeId === bboxId` condition here instead of `activeTest !== bboxTest && activeId === bboxId || bbox.groupId === props.groupId`?